src/target/ext/base/singlethread.rs

```diff
@@ -98,6 +112,39 @@ pub trait SingleThreadOps: Target {
     ) -> TargetResult<(), Self>;
 }
 
+/// Target Extension - [Reverse continue] for single threaded targets.
+///
+/// Reverse continue allows the target to run backwards until it reaches the end
+/// of the replay log.
+///
+/// [Reverse continue]: https://sourceware.org/gdb/current/onlinedocs/gdb/Reverse-Execution.html
+pub trait SingleThreadReverseCont: Target + SingleThreadOps {
+    /// Reverse-continue the target.
+    fn reverse_cont(
+        &mut self,
+        check_gdb_interrupt: &mut dyn FnMut() -> bool,
+    ) -> Result<StopReason<<Self::Arch as Arch>::Usize>, Self::Error>;
+}
+
+define_ext!(SingleThreadReverseContOps, SingleThreadReverseCont);
+
+/// Target Extension - [Reverse stepping] for single threaded targets.
+///
+/// Reverse stepping allows the target to run backwards by one step.
+/// Usually this corresponds to a single instruction, but it can be any metric
+/// defined by the target.
+///
+/// [Reverse stepping]: https://sourceware.org/gdb/current/onlinedocs/gdb/Reverse-Execution.html
+pub trait SingleThreadReverseStep: Target + SingleThreadOps {
+    /// Reverse-step the specified [`Tid`].
+    fn reverse_step(
+        &mut self,
+        check_gdb_interrupt: &mut dyn FnMut() -> bool,
```

Comment:
Unsure if this is strictly necessary, though it's possible that these operations may take some time.

Reply:
It would be weird not to have it.